pkg/agent/qrm-plugins/network/staticpolicy/policy.go

```diff
@@ -429,44 +526,159 @@ func (p *StaticPolicy) Allocate(_ context.Context,
 		general.Errorf("%s", err.Error())
 		return nil, err
 	}
 
-	// TODO fill it with allocated bandwidth quantity
-	return packAllocationResponse(req, p.ResourceName(), 0, resourceAllocationAnnotations)
+	siblingNUMAs, err := machine.GetSiblingNUMAs(selectedNIC.NumaNode, p.agentCtx.CPUTopology)
+	if err != nil {
+		general.Errorf("get siblingNUMAs for nic: %s failed with error: %v. Incorrect NumaNodes in machineState allocationInfo", selectedNIC.Iface, err)
+	}
+
+	// generate the response hint
+	// it could be different from the req.Hint if the affinitive NIC does not have sufficient bandwidth
+	nicPreference, err := checkNICPreferenceOfReq(selectedNIC, req.Annotations)
+	if err != nil {
+		return nil, fmt.Errorf("checkNICPreferenceOfReq for nic: %s failed with error: %v", selectedNIC.Iface, err)
+	}
+
+	respHint := &pluginapi.TopologyHint{
+		Nodes:     siblingNUMAs.ToSliceUInt64(),
+		Preferred: nicPreference,
+	}
+
+	// generate allocationInfo and update the checkpoint accordingly
+	machineState := p.state.GetMachineState()
+	podEntries := p.state.GetPodEntries()
+	allocationInfo := p.state.GetAllocationInfo(req.PodUid, req.ContainerName)
+
+	if allocationInfo != nil {
+		if allocationInfo.Egress >= uint32(reqInt) && allocationInfo.Ingress >= uint32(reqInt) {
+			general.InfoS("already allocated and meet requirement",
+				"podNamespace", req.PodNamespace,
+				"podName", req.PodName,
+				"containerName", req.ContainerName,
+				"bandwidthReq(Mbps)", reqInt,
+				"currentResult(Mbps)", allocationInfo.Egress)
+
+			resp, packErr := packAllocationResponse(req, allocationInfo, req.Hint, resourceAllocationAnnotations)
+			if packErr != nil {
+				general.Errorf("pod: %s/%s, container: %s packAllocationResponse failed with error: %v",
+					req.PodNamespace, req.PodName, req.ContainerName, packErr)
+				return nil, fmt.Errorf("packAllocationResponse failed with error: %v", packErr)
+			}
+			return resp, nil
+		} else {
+			general.InfoS("not meet requirement, clear record and re-allocate",
+				"podNamespace", req.PodNamespace,
+				"podName", req.PodName,
+				"containerName", req.ContainerName,
+				"bandwidthReq(Mbps)", reqInt,
+				"currentResult(Mbps)", allocationInfo.Egress)
+			delete(podEntries, req.PodUid)
+
+			var stateErr error
+			machineState, stateErr = state.GenerateMachineStateFromPodEntries(p.qrmConfig, p.nics, podEntries, p.state.GetReservedBandwidth())
+			if stateErr != nil {
+				general.ErrorS(stateErr, "generateNetworkMachineStateByPodEntries failed",
+					"podNamespace", req.PodNamespace,
+					"podName", req.PodName,
+					"containerName", req.ContainerName,
+					"bandwidthReq(Mbps)", reqInt,
+					"currentResult(Mbps)", allocationInfo.Egress)
+				return nil, fmt.Errorf("generateNetworkMachineStateByPodEntries failed with error: %v", stateErr)
+			}
+		}
+	}
+
+	// update the machineState
+	machineState[selectedNIC.Iface].EgressState.Allocated += uint32(reqInt)
```

Comment:
not to modify machineState directly,
use GenerateMachineStateFromPodEntries instead.

it's helpful to keep consistency between podEntries and machineState.